pkg/controller/sidecarset/sidecarset_controller.go

```diff
@@ -59,6 +62,11 @@ func Add(mgr manager.Manager) error {
 	if !utildiscovery.DiscoverGVK(controllerKind) {
 		return nil
 	}
+	if !utildiscovery.DiscoverGVK(appsv1alpha1.SchemeGroupVersion.WithKind("ImagePullJob")) ||
+		!utilfeature.DefaultFeatureGate.Enabled(features.KruiseDaemon) ||
+		!utilfeature.DefaultFeatureGate.Enabled(features.PreDownloadImageForInPlaceUpdate) {
```

Comment:
Can you add new feature gate PreDownloadImageForSidecarSet?

Reply:
Sure, let me try on the new feature.

Reply:
I have add the `apps.kruise.io/image-predownload-min-updated-ready-pods` to make sure the new image starting pre-download after a few new Pods have been updated ready.

Reply:
fix it